Demo/UIKit/CustomComponent/CustomComponentPresenter.swift

```diff
@@ -82,4 +133,37 @@ extension CustomComponentPresenter: PaymentComponentDelegate {
         alertViewController.addAction(doneAction)
         (view as? UIViewController)?.present(alertViewController, animated: true)
     }
+
+    private func stopLoading() {
+        view?.stopActivityIndicator()
+        cardComponent?.stopLoadingIfNeeded()
+    }
+}
+
+extension CustomComponentPresenter: ActionComponentDelegate {
+    
+    internal func didFail(with error: Error, from component: ActionComponent) {
+        handlePayment(error: error)
+    }
+
+    internal func didComplete(from component: ActionComponent) { /* Empty implementation */ }
+
+    internal func didProvide(_ data: ActionComponentData, from component: ActionComponent) {
+        let request = PaymentDetailsRequest(
+            details: data.details,
+            paymentData: data.paymentData,
+            merchantAccount: ConfigurationConstants.current.merchantAccount
+        )
+        apiClient.perform(request) { [weak self] result in
+            self?.handlePayment(result: result)
+        }
+    }
+}
+
+extension CustomComponentPresenter: PresentationDelegate {
+    
+    func present(component: any PresentableComponent) {
+        let viewController = component.viewController
+        (view as? UIViewController)?.present(viewController, animated: true)
```

Comment:
`view` is an abstraction (protocol). Since the `CustomComponentViewController` conforms to it, I decided it's safe to cast. I'm aware it's not the best approach but since this is just demo code and will go away after the task I chose this approach instead of defining more methods to present. To sum up, it won't fail.

Reply:
I changed it to avoid the cast. It should be better, it was a small change.